### PR TITLE
Add expvar for custom metrics to DCA and refactor status package.

### PR DESCRIPTION
**Pull request by devonboyer**
### What does this PR do?

- Adds the `render` package that includes the shared code for rendering go templates for the `status` command (extracted from the `status` package).
- Adds expvar for custom metrics to the DCA

```
$ curl localhost:5000/debug/vars

...
"custommetrics": {"External":{"Total":1,"Valid":0}}
...
```
```
--- FAIL: TestFormatHPAStatus (0.00s)
	render_test.go:98: Rendering "error"...
		
		Custom Metrics Provider
		=======================
		  Error: This is an error
		  
	render_test.go:98: Rendering "store error"...
		
		Custom Metrics Provider
		=======================
		  ConfigMap name: datadog-hpa
		  
		  Error: This is an error
		  
	render_test.go:98: Rendering "list error"...
		
		Custom Metrics Provider
		=======================
		  ConfigMap name: datadog-hpa
		  
		  External Metrics
		  ----------------
		    Error: This is an error
		    
	render_test.go:98: Rendering "no metrics"...
		
		Custom Metrics Provider
		=======================
		  ConfigMap name: datadog-hpa
		  
		  External Metrics
		  ----------------
		    Total: 2
		    Valid: 1
		    
	render_test.go:98: Rendering "one metric"...
		
		Custom Metrics Provider
		=======================
		  ConfigMap name: datadog-hpa
		  
		  External Metrics
		  ----------------
		    Total: 2
		    Valid: 1
		    
		    hpa:
		    - name: hpa
		    - namespace: default
		    labels:
		    - foo: bar
		    metricName: metric1
		    ts: 1.533262651e&#43;09
		    valid: true
		    value: 10
		    
	render_test.go:98: Rendering "multiple metrics"...
		
		Custom Metrics Provider
		=======================
		  ConfigMap name: datadog-hpa
		  
		  External Metrics
		  ----------------
		    Total: 2
		    Valid: 1
		    
		    hpa:
		    - name: hpa
		    - namespace: default
		    labels:
		    - foo: bar
		    metricName: metric1
		    ts: 1.533262651e&#43;09
		    valid: true
		    value: 10
		    
		    hpa:
		    - name: hpa
		    - namespace: default
		    labels:
		    - foo: bar
		    metricName: metric2
		    ts: 1.533262651e&#43;09
		    valid: false
		    value: 10
```

### Motivation

The `status` package contained the status code for both the Agent as well as the client and server status for the Cluster Agent. It was starting to get very confusing and hard to follow.

### CODE REVIEW

File: pkg/status/status.go [hunk 1]
```diff
@@ -52,14 +52,14 @@ func GetStatus() (map[string]interface{}, error) {
...
-	stats["time"] = now.Format(timeFormat)
+	stats["time"] = now.Format(render.TimeFormat)
 
 	stats["JMXStatus"] = GetJMXStatus()
 
 	stats["logsStats"] = logs.GetStatus()
 
 	if config.Datadog.GetBool("cluster_agent.enabled") {
-		stats["clusterAgentStatus"] = getDCAStatus()
+		stats["clusterAgentStatus"] = clusteragent.GetStatus()
```
Comment (devonboyer) [inline pkg/status/status.go]: This `clusteragent.GetStatus()` is the client-side status which lives in the `util/clusteragent` package.

File: pkg/clusteragent/custommetrics/stats.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	v, ok := status["External"]
+	if !ok {
+		return stats
+	}
+	externalStatus, ok := v.(map[string]interface{})
+	if !ok {
+		return stats
+	}
+	stats.External.Total, _ = externalStatus["Total"].(int)
+	stats.External.Valid, _ = externalStatus["Valid"].(int)
```
Comment (devonboyer) [inline pkg/clusteragent/custommetrics/stats.go]: I tested in a go playground that this will not panic if conversion fails.

**Review by hush-hush — COMMENTED** [review]
Quick questions before a more in depth review:

I thought the goal of refactoring the status package was to merge `Dockerfiles/cluster-agent/dist/templates` and `pkg/status/dist/templates`. Is this part of another PR ?

Also side question why extract the `render` package from the `status` and have 2 of them (one in the `clusteragent` and one in `pkg`) ?

My understanding was that we wanted to merge everything in one package that would be able to render the status page for everybody.

Side note: love the `GetStatus` for each package. Maybe having a `register` method in the status package where each module would add himself at startup would make a more dynamic status page that fit the agent, clusteragent and dogstatsd ?

**Comment by devonboyer** [comment]
I didn't intend to merge `Dockerfiles/cluster-agent/dist/templates` and `pkg/status/dist/templates` but that may be a good idea.

The idea of registering each `GetStatus` function is interesting and it might solve all my concerns with separating out the shared code into the `render` package.

I'll give it another look to see if I can do the registering idea to clean it up even more.

**Comment by devonboyer** [comment]
I am closing this to split it into multiple smaller PRs.
